src/test/java/com/keronic/majestik/ast/IdentityExpressionNodeTest.java

```diff
@@ -0,0 +1,74 @@
+package com.keronic.majestik.ast;
+
+import module java.base;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import org.junit.jupiter.api.Test;
+
+class IdentityExpressionNodeTest extends NodeTest {
+  @Test
+  void testEquals() {
+    final var node1 = new IdentityExpressionNode(new VariableNode(0), new VariableNode(1));
+    final var node2 = new IdentityExpressionNode(new VariableNode(0), new VariableNode(1));
+    final var node3 = new IdentityExpressionNode(new VariableNode(0), new VariableNode(2));
+
+    assertEquals(node1, node2);
+
+    assertNotEquals(node1, node3);
+    assertNotEqualsNull(node2);
+  }
+
```

Comment:
`assertNotEqualsNull` is a custom assertion which uses the equality test to compare to null (instead of identity). Name is therefore clear and concise.